From baab72dfd8804f78ab47d6fc4ff92beec28cd6c7 Mon Sep 17 00:00:00 2001
From: wu xia <xia.wu@intel.com>
Date: Fri, 18 Sep 2015 20:42:03 +0300
Subject: [PATCH 0941/1240] media: v4l: Rework the bayer vector format
 definitions a bit

Use V32 to denote the vector length. Also remove the 16-bit formats as
they're unused at the moment.

Signed-off-by: Sakari Ailus <sakari.ailus@linux.intel.com>
Signed-off-by: wu xia <xia.wu@intel.com>
---
 include/uapi/linux/videodev2.h | 19 +++++++++++--------
 1 file changed, 11 insertions(+), 8 deletions(-)

diff --git a/include/uapi/linux/videodev2.h b/include/uapi/linux/videodev2.h
index 4227c57..12811ea 100644
--- a/include/uapi/linux/videodev2.h
+++ b/include/uapi/linux/videodev2.h
@@ -615,10 +615,10 @@ struct v4l2_pix_format {
 #define V4L2_PIX_FMT_SRGGB16 v4l2_fourcc('R', 'G', '1', '6') /* 16  RGRG.. GBGB.. */
 
 /* Raw bayer vector formats. */
-#define V4L2_PIX_FMT_SBGGR8V32         v4l2_fourcc('b', 'V', '0', 'A')
-#define V4L2_PIX_FMT_SGBRG8V32         v4l2_fourcc('b', 'V', '0', 'B')
-#define V4L2_PIX_FMT_SGRBG8V32         v4l2_fourcc('b', 'V', '0', 'C')
-#define V4L2_PIX_FMT_SRGGB8V32         v4l2_fourcc('b', 'V', '0', 'D')
+#define V4L2_PIX_FMT_SBGGR8_16V32      v4l2_fourcc('b', 'V', '0', 'A')
+#define V4L2_PIX_FMT_SGBRG8_16V32      v4l2_fourcc('b', 'V', '0', 'B')
+#define V4L2_PIX_FMT_SGRBG8_16V32      v4l2_fourcc('b', 'V', '0', 'C')
+#define V4L2_PIX_FMT_SRGGB8_16V32      v4l2_fourcc('b', 'V', '0', 'D')
 #define V4L2_PIX_FMT_SBGGR10V32                v4l2_fourcc('b', 'V', '0', 'E')
 #define V4L2_PIX_FMT_SGBRG10V32                v4l2_fourcc('b', 'V', '0', 'F')
 #define V4L2_PIX_FMT_SGRBG10V32                v4l2_fourcc('b', 'V', '0', 'G')
@@ -627,10 +627,13 @@ struct v4l2_pix_format {
 #define V4L2_PIX_FMT_SGBRG12V32                v4l2_fourcc('b', 'V', '0', 'J')
 #define V4L2_PIX_FMT_SGRBG12V32                v4l2_fourcc('b', 'V', '0', 'K')
 #define V4L2_PIX_FMT_SRGGB12V32                v4l2_fourcc('b', 'V', '0', 'L')
-#define V4L2_PIX_FMT_SBGGR16V32                v4l2_fourcc('b', 'V', '0', 'M')
-#define V4L2_PIX_FMT_SGBRG16V32                v4l2_fourcc('b', 'V', '0', 'N')
-#define V4L2_PIX_FMT_SGRBG16V32                v4l2_fourcc('b', 'V', '0', 'O')
-#define V4L2_PIX_FMT_SRGGB16V32                v4l2_fourcc('b', 'V', '0', 'P')
+
+/* BEGIN remove once the user space has been updated */
+#define V4L2_PIX_FMT_SBGGR8V32 v4l2_fourcc('b', 'V', '0', 'A')
+#define V4L2_PIX_FMT_SGBRG8V32 v4l2_fourcc('b', 'V', '0', 'B')
+#define V4L2_PIX_FMT_SGRBG8V32 v4l2_fourcc('b', 'V', '0', 'C')
+#define V4L2_PIX_FMT_SRGGB8V32 v4l2_fourcc('b', 'V', '0', 'D')
+/* END remove once the user space has been updated */
 
 /* YUV vector formats. */
 #define V4L2_PIX_FMT_UYVY_V32          v4l2_fourcc('y', 'V', '3', '2')
-- 
2.7.4

